llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -22514,14 +22514,22 @@ static SDValue performSTNT1Combine(SDNode *N, SelectionDAG &DAG) {
 ///   movi v0.2d, #0
 ///   str q0, [x0]
 ///
-static SDValue replaceZeroVectorStore(SelectionDAG &DAG, StoreSDNode &St) {
+static SDValue replaceZeroVectorStore(SelectionDAG &DAG, StoreSDNode &St,
+                                      AArch64Subtarget const &Subtarget) {
   SDValue StVal = St.getValue();
   EVT VT = StVal.getValueType();
 
   // Avoid scalarizing zero splat stores for scalable vectors.
   if (VT.isScalableVector())
     return SDValue();
 
+  // Do not replace the FP store when it could result in a streaming memory
+  // hazard.
+  if (VT.getVectorElementType().isFloatingPoint() &&
+      Subtarget.getStreamingHazardSize() > 0 &&
+      (Subtarget.isStreaming() || Subtarget.isStreamingCompatible()))
```

Comment:
Can this call `canUseIntLoadStoreForFloatValues` ?

Reply:
Yep, done :+1: